expression/builtin_math_test.go

```diff
@@ -231,3 +231,29 @@ func (s *testEvaluatorSuite) TestConv(c *C) {
 		c.Assert(r, Equals, t.ret)
 	}
 }
+
+func (s *testEvaluatorSuite) TestSign(c *C) {
+	defer testleak.AfterTest(c)()
+
+	for _, t := range []struct {
+		num interface{}
+		ret interface{}
+	}{
+		{nil, nil},
+		{1, 1},
+		{0, 0},
+		{-1, -1},
+		{0.4, 1},
+		{-0.4, -1},
+		{"1", 1},
+		{"-1", -1},
+		{"1a", 1},
```

Comment:
You mean `0x1a` and `-0x1a` here?
Which type will `1a` be treated as in `types.MakeDaums`?